### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by jon-malave** [node-cron/node-cron#213]
**Use Case:**
I am currently using node-cron to schedule text message notifications. We need to create a cron job that sends text message notifications to end-users at a set hourly interval; then destroy that cron job after an expiration date.

**How to Reproduce:**
- Create a cron job that handles sending text message notification every X number of hours.
- Create another cron job that handles destroying the re-occurring cron job above, and it-self.
- Both of the above steps work as expected (if we check the status of the cron jobs using the `.getStatus()` method they return undefined because they no longer exist).
- Sometimes the cron job that handles sending text message notifications still exist in memory (even after being destroyed using the `.destroy()` method) and continues to send notifications. 
- We have no way of stopping the notifications or destroying the cron job from within the running code unless we restart the application. Restarting the application every-time a single cron job failed to be properly destroyed is not an ideal solution.

**Comment by jon-malave** [comment]
@jonathangaldino @TDola @oferze @merencia @t-bonatti hopefully one of you can help!? This issue is blocking us from pushing our current application into production. Please advise if possible... would be so appreciated!

**Comment by TDola** [comment]
Did you call stop() before destroying it?

**Comment by jon-malave** [comment]
> Did you call stop() before destroying it?

No I am not calling stop. I am just using `destory()` as destroy method is suppose to stop and destroy the job. The destroy method does work, as the above reported issue does not happen all the time! Sometimes my cron jobs will be stopped and destroyed properly, sometimes not. One out of every few cron jobs I schedule won't properly destroy and continue running memory performing it's task of sending text message notifications.

**Comment by TDola** [comment]
I am not sure then.  I always call stop and then destroy the array that held all cron objects and replace with a new list.  No memory leaks and it is really stable.

**Comment by jon-malave** [comment]
I guess I can try to use `.stop()` right before I call `.destroy()` and see what the results are...

**Comment by jon-malave** [comment]
@TDola so using the `.stop()` then `.destroy()` seems to have helped in my last test scheduling over 15 jobs, everything started and stopped as expected. 

Although I did another test yesterday, and I ended up with 6 jobs that did not end when it was suppose to. 

I restarted my node-cron instance, and tried again and it's working now. I'm curious to know how much of this is possibly my Hardware? I saw some significant improvements when I upgraded my Google Cloud VM Instance from a 1vCPU to a 4vCPU, but i'm wondering if I still need even more processing Power than this to help eliminate an errors in execution? Especially being that Cron Jobs are very time sensitive. If a Job takes 1 second to long it could effect the execution of the next Job potentially...

**Comment by TDola** [comment]
You could childProcess.spawn your jobs onto another thread and not hold up the event loop on your main thread.
I do this, and also because mine runs on a windows machine, I run this line
`exec('wmic process where "ProcessId=' + process.pid + '" CALL setpriority 256');`
This sets the priority of the process to the highest possible so the event loop never gets stuck waiting for IO as the operating system spins up another node instance for each spawn.

**Comment by jon-malave** [comment]
> You could childProcess.spawn your jobs onto another thread and not hold up the event loop on your main thread.
> I do this, and also because mine runs on a windows machine, I run this line
> `exec('wmic process where "ProcessId=' + process.pid + '" CALL setpriority 256');`
> This sets the priority of the process to the highest possible so the event loop never gets stuck waiting for IO as the operating system spins up another node instance for each spawn.

Interesting! I have never used child process before... OK, I'll start reading up on this and hopefully can implement and helps my current situation. Thanks a bunch!!

**Comment by ihor-himatov** [comment]
@jon-malave did child process help you to solve current issue with crons?

**Comment by ihor-himatov** [comment]
@jon-malave did child process help you to solve current issue with crons?

**Comment by jonmalave** [comment]
> @jon-malave did child process help you to solve current issue with crons?

Yep, it did solve my issue. It works exactly as @TDola described.
